models/models.py

```diff
@@ -0,0 +1,95 @@
+from sqlalchemy.orm import relationship, backref
+from sqlalchemy.ext.declarative import declarative_base, declared_attr
+from sqlalchemy import Column, Integer, String, ForeignKey, \
+    Date, Float, Sequence
+
+STRLEN = 50
+
+
+class Base:
+    @declared_attr
+    def __tablename__(cls):
+        return cls.__name__.lower()
+
+    @declared_attr
+    def id(cls):
+        return Column(Integer, Sequence(cls.__name__.lower() + 'id_seq'), primary_key=True)
+
+Base = declarative_base(cls=Base)
+
+
+class Category(Base):
+    name = Column(String(STRLEN))
+    description = Column(String(STRLEN))
+    parent_id = Column(Integer, ForeignKey('category.id'), nullable=True)
+    children = relationship(
+        'Category',
+        backref=backref('parent', remote_side=[id])
+    )
+    rates = relationship(
+        'Rates',
+        backref=backref('category')
```

Comment:
http://docs.sqlalchemy.org/en/latest/orm/basic_relationships.html#one-to-many
>Alternatively, the backref option may be used on a single relationship() instead of using back_populates: